### PR TITLE
Reply feature for doctor notes

### LINKED ISSUE #7706: Reply function on Doctor's Note
Develop an option to "reply" against a message on the doctor's note. Once user clicks on the reply option under a message, they must be able to add text message and send it as a reply. 

Eg: Whatsapp replies

This feature will in future allow us to understand how a suggestion added to a the doctor's note is received.

### PR DESCRIPTION
**Pull request by UdaySagar-Git**
## Proposed Changes

- Fixes #7706 

Backend PR : https://github.com/coronasafe/care/pull/2160

[Screencast from 2024-05-14 03-25-10.webm](https://github.com/coronasafe/care_fe/assets/111575806/274eeac7-08ef-4e61-a798-6875709efe78)

image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by nihal467** [comment]
@UdaySagar-Git fix the lint and test in the backend PR

File: src/Components/Facility/ConsultationDoctorNotes/index.tsx [hunk 1]
```diff
@@ -38,6 +39,9 @@ const ConsultationDoctorNotes = (props: ConsultationDoctorNotesProps) => {
   const [facilityName, setFacilityName] = useState("");
   const [patientName, setPatientName] = useState("");
   const [focused, setFocused] = useState(false);
+  const [reply_to, setReplyTo] = useState<PaitentNotesReplyModel | undefined>(
+    undefined,
+  );
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationDoctorNotes/index.tsx:44]: `useState` will infer if the type is ` | undefined` or not with `T`

```suggestion
  const [replyTo, setReplyTo] = useState<PaitentNotesReplyModel>();
```

File: src/Components/Facility/PatientNotesSlideover.tsx [hunk 2]
```diff
@@ -34,6 +35,9 @@ export default function PatientNotesSlideover(props: PatientNotesProps) {
   const [patientActive, setPatientActive] = useState(true);
   const [reload, setReload] = useState(false);
   const [focused, setFocused] = useState(false);
+  const [reply_to, setReplyTo] = useState<PaitentNotesReplyModel | undefined>(
+    undefined,
+  );
```
Comment (rithviknishad) [inline src/Components/Facility/PatientNotesSlideover.tsx:40]: same here...

also consider abstracting the repeated logic to a common component. (To be done in upcoming PRs for enhancing doctor notes)

**Review by rithviknishad — APPROVED** [review]
LGTM

**Comment by nihal467** [comment]
functionality looks good, approved it , UI need to be improved as discussed in slack ( Ref : https://rebuildearth.slack.com/archives/C010GQC5UE9/p1715235428755999?thread_ts=1715201296.462099&cid=C010GQC5UE9 )

**Comment by UdaySagar-Git** [comment]
@nihal467 sure , will improve in upcomming pr's !

**Comment by bodhish** [comment]
@nihal467 @UdaySagar-Git Whats the status here?

**Comment by UdaySagar-Git** [comment]
@bodhish it is a part of https://github.com/ohcnetwork/care_fe/pull/8016,
updated the pr, the main pr has much better ui , waiting for some one's review

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**